File: apps/ios/Photon/Models/UserSession.swift
//
//  UserSession.swift
//  Photon
//
//  Created by Metin TASCI on 14.08.2026.
//

import Foundation

/// Represents the active user authentication state and profile metadata.
public struct UserSession: Equatable, Codable, Sendable {
    public let uid: String
    public let email: String?
    public let displayName: String?
    public let photoURL: URL?
    public let providerId: String?
    public let isAnonymous: Bool
    
    public init(
        uid: String,
        email: String? = nil,
        displayName: String? = nil,
        photoURL: URL? = nil,
        providerId: String? = nil,
        isAnonymous: Bool = false
    ) {
        self.uid = uid
        self.email = email
        self.displayName = displayName
        self.photoURL = photoURL
        self.providerId = providerId
        self.isAnonymous = isAnonymous
    }
}

File: apps/android/app/src/main/java/com/alafteknoloji/photon/models/UserSession.kt
package com.alafteknoloji.photon.models

import android.graphics.Bitmap
import android.net.Uri

/**
 * Model representing the authenticated user's session in Photon.
 * 100% Parity with iOS UserSession.swift.
 */
data class UserSession(
    val uid: String,
    val email: String? = null,
    val displayName: String = "Phonla Üyesi",
    val photoUrl: String? = null,
    val phoneNumber: String? = null,
    val providerId: String = "Firebase",
    val isAnonymous: Boolean = false
)

/**
 * In-memory container for a loaded photo in the Photon Editor.
 * 100% Parity with iOS LoadedPhoto.swift.
 */
data class LoadedPhoto(
    val sourceUri: Uri? = null,
    val originalBitmap: Bitmap,
    val previewBitmap: Bitmap,
    val width: Int = originalBitmap.width,
    val height: Int = originalBitmap.height,
    val filename: String = "photon_photo.jpg"
)
